assembly/utils.ts

```diff
@@ -395,83 +395,87 @@ export function balanceDuration(
     nanoseconds as i64
   );
 
+  let nanosecondsI64: i64 = 0;
+  let microsecondsI64: i64 = 0;
+  let millisecondsI64: i64 = 0;
+  let secondsI64: i64 = 0;
+  let minutesI64: i64 = 0;
+  let hoursI64: i64 = 0;
+  let daysI64: i64 = 0;
+
   if (
     largestUnit >= TimeComponent.years &&
     largestUnit <= TimeComponent.days
   ) {
     const _ES$NanosecondsToDays = nanosecondsToDays(durationNs);
-    days        = _ES$NanosecondsToDays.days;
-    nanoseconds = _ES$NanosecondsToDays.nanoseconds;
+    daysI64        = _ES$NanosecondsToDays.days;
+    nanosecondsI64 = _ES$NanosecondsToDays.nanoseconds;
   } else {
-    days = 0;
+    daysI64 = 0;
+    nanosecondsI64 = durationNs
   }
 
-  const sig = sign(nanoseconds);
-  nanoseconds = abs(nanoseconds);
-  microseconds = 0;
-  milliseconds = 0;
-  seconds = 0;
-  minutes = 0;
-  hours = 0;
+  const sig = sign(nanosecondsI64);
```

Comment:
```suggestion
  const sig = sign(nanoseconds);
```

Reply:
Does casting sig to `i32` change something?

Reply:
See next suggestion. Main idea is avoiding 64-bit arithmetic as much as possible due to some wasm engines or some 32-bit platforms may be significantly slower with 64-bit arithmetic

Reply:
Ohh, I see.